src/python/pants/backend/go/util_rules/sdk.py

```diff
@@ -125,6 +132,29 @@ async def setup_go_sdk_process(
         "__PANTS_GO_SDK_CACHE_KEY": f"{goroot.full_version}/{goroot.goos}/{goroot.goarch}",
     }
 
+    immutable_input_digests = {}
+
+    # Add path to additional tools, such as git, that may be needed by the go tool
+    if golang_env_aware.extra_tools:
+        extra_tools = await Get(
+            BinaryShims,
+            BinaryShimsRequest,
+            BinaryShimsRequest.for_binaries(
+                *golang_env_aware.extra_tools,
+                rationale="allow additional tools for go tools",
+                search_path=search_path,
+            ),
+        )
+        # Append path to additional tools
+        if "PATH" in env:
+            env["PATH"] = f"{extra_tools.path_component}:{env['PATH']}"
+        else:
+            env["PATH"] = extra_tools.path_component
+        immutable_input_digests = {
```

Comment:
More compact is: `immutable_input_digests.update(extra_tools.immutable_input_digests)`